tests/lib/rules/no-unused-prop-types.js

```diff
@@ -1208,6 +1208,96 @@ ruleTester.run('no-unused-prop-types', rule, {
       ].join('\n'),
       options: [{skipShapeProps: true}],
       parser: 'babel-eslint'
+    }, {
+      // Destructured props in componentWillReceiveProps shouldn't throw errors
+      code: [
+        'class Hello extends Component {',
+        '  static propTypes = {',
+        '    something: PropTypes.bool',
+        '  }',
+        '  componentWillReceiveProps (nextProps) {',
+        '    const {something} = nextProps;',
+        '    doSomething(something);',
+        '  }',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // Destructured props in the constructor shouldn't throw errors
+      code: [
+        'class Hello extends Component {',
+        '  static propTypes = {',
+        '    something: PropTypes.bool',
+        '  }',
+        '  constructor (props) {',
+        '    super(props);',
+        '    const {something} = props;',
```

Comment:
can we also test all these cases with the destructuring being done inline in the signature? ie, `constructor ({ something }) {`?